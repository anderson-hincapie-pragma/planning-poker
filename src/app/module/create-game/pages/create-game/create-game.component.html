<app-create-game-template
    [labelText]="labelText"
    [inputType]="inputType"
    [inputControl]="nameControl"
    [actionButtonText]="actionButtonText"
    [buttonDisabled]="createGameControl.invalid"
    (actionButtonClick)="createGame($event)">
</app-create-game-template>