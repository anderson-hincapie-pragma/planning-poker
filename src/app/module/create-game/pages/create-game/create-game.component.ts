import { Component, Input } from '@angular/core';
import { FormBuilder, FormControl, FormGroup, Validators } from '@angular/forms';

@Component({
  templateUrl: './create-game.component.html',
  styleUrls: ['./create-game.component.scss']
})
export class CreateGameComponent {

  public labelText: string = 'Nombra la partida';
  public inputType: 'email' | 'number' | 'search' | 'text' = 'text';
  public createGameControl: FormGroup = this.formBuilder.group({
    name: [{ value: null, disabled: false}, [Validators.required]]
  });
  get nameControl() {
    return this.createGameControl.get('name') as FormControl;
  }
  public actionButtonText: string = 'Crear partida';

  constructor(
    private formBuilder: FormBuilder
  ) {}

  createGame(event: MouseEvent) {
    if (this.createGameControl.invalid) {
      return;
    }

    console.log("create game>>>");
  }

}
