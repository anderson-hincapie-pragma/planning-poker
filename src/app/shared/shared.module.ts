import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ActionButtonComponent } from './components/atoms/action-button/action-button.component';
import { InputComponent } from './components/atoms/input/input.component';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { FormInputComponent } from './components/molecules/form-input/form-input.component';
import { InputCardComponent } from './components/organisms/input-card/input-card.component';
import { CreateGameTemplateComponent } from './components/templates/create-game-template/create-game-template.component';



@NgModule({
  declarations: [
    ActionButtonComponent,
    InputComponent,
    FormInputComponent,
    InputCardComponent,
    CreateGameTemplateComponent
  ],
  imports: [
    CommonModule,
    ReactiveFormsModule
  ],
  exports: [
    ActionButtonComponent,
    InputComponent,
    FormInputComponent,
    InputCardComponent,
    CreateGameTemplateComponent
  ]
})
export class SharedModule { }
