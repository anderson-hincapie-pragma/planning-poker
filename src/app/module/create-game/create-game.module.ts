import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';

import { CreateGameRoutingModule } from './create-game-routing.module';
import { CreateGameComponent } from './pages/create-game/create-game.component';
import { SharedModule } from '@shared/shared.module';

@NgModule({
  declarations: [
    CreateGameComponent
  ],
  imports: [
    CommonModule,
    CreateGameRoutingModule,
    SharedModule
  ]
})
export class CreateGameModule { }
