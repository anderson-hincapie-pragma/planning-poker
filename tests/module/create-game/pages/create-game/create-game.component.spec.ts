import { ComponentFixture, TestBed } from '@angular/core/testing';

import { CreateGameComponent } from '@module/create-game/pages/create-game/create-game.component';
import { SharedModule } from '@shared/shared.module';

describe('CreateGameComponent', () => {
  let component: CreateGameComponent;
  let fixture: ComponentFixture<CreateGameComponent>;
  let compiled: HTMLElement;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [SharedModule],
      declarations: [CreateGameComponent]
    });
    fixture = TestBed.createComponent(CreateGameComponent);
    compiled = fixture.nativeElement;
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    expect(compiled).toMatchSnapshot();
  })

});
