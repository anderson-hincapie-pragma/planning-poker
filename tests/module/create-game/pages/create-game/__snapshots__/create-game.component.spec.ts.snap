// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateGameComponent should match snapshot 1`] = `
<div
  id="root1"
>
  <app-create-game-template>
    <div
      class="t-create-game"
    >
      <app-input-card
        class="t-create-game__input-card"
      >
        <div
          class="o-input-card"
        >
          <app-form-input
            class="o-input-card__form-input"
          >
            <div
              class="m-form-input"
            >
              <label
                class="m-form-input__label"
              >
                Nombra la partida
              </label>
              <app-input
                class="m-form-input__input"
              >
                <input
                  class="a-input ng-untouched ng-pristine ng-invalid"
                  placeholder=""
                  type="text"
                />
              </app-input>
            </div>
          </app-form-input>
          <app-action-button
            class="o-input-card__action-button"
          >
            <button
              class="a-action-button"
              data-testid="action-button"
              disabled=""
            >
              Crear partida
            </button>
          </app-action-button>
        </div>
      </app-input-card>
    </div>
  </app-create-game-template>
</div>
`;
